packages/openapi-v3-types/src/openapi-v3-spec-types.ts

```diff
@@ -0,0 +1,395 @@
+ * Specification.
+ */
+export interface OAS3SchemaObject extends ISpecificationExtension {
+  nullable?: boolean;
+  discriminator?: DiscriminatorObject;
+  readOnly?: boolean;
+  writeOnly?: boolean;
+  xml?: XmlObject;
+  externalDocs?: ExternalDocumentationObject;
+  example?: ExtensionValue;
+  examples?: ExtensionValue[];
+  deprecated?: boolean;
+
+  type?: string;
+  allOf?: (SchemaObject | ReferenceObject)[];
+  oneOf?: (SchemaObject | ReferenceObject)[];
+  anyOf?: (SchemaObject | ReferenceObject)[];
+  not?: SchemaObject | ReferenceObject;
+  items?: SchemaObject | ReferenceObject;
+  properties?: {[propertyName: string]: SchemaObject | ReferenceObject};
+  additionalProperties?: SchemaObject | ReferenceObject;
+  description?: string;
+  format?: string;
+  default?: ExtensionValue;
+}
+
+/**
+ * JSON type - This is part of the Schema object.
+ * The following properties are taken directly from the JSON Schema
+ * definition and follow the same specifications.
+ * See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schema-object
+ */
+
+export type JSONType = {
+  title?: string;
+  multipleOf?: number;
+  maximum?: number;
+  exclusiveMaximum?: number;
+  minimum?: number;
+  exclusiveMinimum?: number;
+  maxLength?: number;
+  minLength?: number;
+  // (This string SHOULD be a valid regular expression, according to the ECMA 262 regular expression dialect)
+  pattern?: string;
+  maxItems?: number;
+  minItems?: number;
+  uniqueItems?: boolean;
+  maxProperties?: number;
+  minProperties?: number;
+  enum?: Array<ExtensionValue>;
+};
+
+/**
+ * Describes a single request body.
+ * <p>Specification:
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#request-body-object
+ */
+export interface RequestBodyObject extends ISpecificationExtension {
+  description?: string;
+  content: ContentObject;
+  required?: boolean;
+}
+
+/**
+ * Describes an object of multiple content types.
+ * For example:
+ * ```js
+ * {
+ *   'application/json': {
+ *     schema: {...schemaObjectSpec}
+ *   },
+ *   'application/text': {
+ *     schema: {...schemaObjectSpec}
+ *   }
+ * }
+ * ```
+ */
+export interface ContentObject {
+  [mediatype: string]: MediaTypeObject;
+}
+
+/**
+ * Each Media Type Object provides schema and examples for the media type
+ * identified by its key.
+ * <p>Specification:
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#media-type-object
+ */
+export interface MediaTypeObject extends ISpecificationExtension {
+  schema?: SchemaObject | ReferenceObject;
+  examples?: [ExampleObject | ReferenceObject];
+  example?: ExampleObject | ReferenceObject;
+  encoding?: EncodingObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingObject extends ISpecificationExtension {
+  // [property: string]: EncodingPropertyObject;
```

Comment:
No commented code.

Reply:
hmm, I need to double check those bad copies